File: Day1/Day1.fs
namespace FSharp.Code

module Day1 =
    let countGreaterValues (nums: seq<int>) =
        Seq.pairwise nums
        |> Seq.filter (fun (a, b) -> a < b)
        |> Seq.length

    let countSlidingGreaterValues (nums: seq<int>) =
        Seq.windowed 3 nums
        |> Seq.map Seq.sum
        |> countGreaterValues

File: AdventRunner/AdventDays/Day1.cs
using System.Collections.Generic;
using System.Linq;
using System.Windows;
using System.Windows.Controls;
using AdventRunner.Helpers;

namespace AdventRunner.AdventDays
{
    public class Day1 : Day
    {
        public static void Run()
        {
            List<int> numbers = GetNewlineSeparatedFileContents(nameof(Day1))
                    .Select(int.Parse)
                    .ToList();

            int greaterDepthCount = FSharp.Code.Day1.countGreaterValues(numbers);
            int slidingGreaterDepthCount = FSharp.Code.Day1.countSlidingGreaterValues(numbers);
            
            var textBox = Application.Current.MainWindow.FindChild<TextBox>("MainTextBox");
            textBox.Text = $"Part 1: {greaterDepthCount}\r\nPart 2: {slidingGreaterDepthCount}";
        }
    }
}